File: frontend/src/types/player.ts
import type { SportType } from "./common";

export interface PlayerSummary {
  id: number;
  name: string;
  team: string;
  team_id: number | null;
  position: string | null;
  photo_url: string | null;
  league_id: number | null;
  league_name: string | null;
  sport: SportType;
}

export interface League {
  id: number;
  name: string;
  country: string;
  season: number;
}

export interface LeagueSelection {
  id: number;
  season: number;
  name: string;
}

File: backend/app/models/player.py
from app.models.common import SportType
from pydantic import BaseModel, Field


class PlayerSummary(BaseModel):
    id: int
    name: str
    team: str
    team_id: int | None = None
    position: str | None = None
    photo_url: str | None = None
    league_id: int | None = None
    league_name: str | None = None
    sport: SportType = SportType.SOCCER
